src/crypto/verifier.h

```diff
@@ -138,9 +129,6 @@ namespace crypto
       size_t sig_size,
       MDType md_type = MDType::NONE)
     {
-      if (md_type == MDType::NONE)
-        md_type = this->md_type;
-
       return public_key->verify_hash(hash, hash_size, sig, sig_size, md_type);
```

Comment:
There may be a problem with `md_type == None` here. I think it works fine with OpenSSL, but the corresponding mbedTLS version won't even attempt to verify anything and return an error. Their docs say `md_alg may be MBEDTLS_MD_NONE, only if hash_len != 0`, but that's incorrect, it will throw an error even when `hash_len != 0`. Could you try running the tests with `-DCRYPTO_PROVIDER_IS_MBEDTLS`?

Reply:
I believe we always grab a valid `md_type` later, in both the OpenSSL and mbedTLS implementations

```
    if (md_type == MDType::NONE)
    {
      md_type = get_md_for_ec(get_curve_id());
    }
```

I've tried several unit tests (`channels_test`, `crypto_test`, `tls_test`, `secret_sharing_test`) with `CRYPTO_PROVIDER_IS_MBEDTLS` and they're all happy.